src/modules/RokulRunnings.ts

```diff
@@ -719,8 +721,48 @@ export class RokulRunnings {
     responseObject.forEach(element => {
       const elementName = Object.keys(element)[0];
       const childElement = element[elementName] as AppUIResponseObject;
-      elementsArray.push({ [elementName]: childElement.attributes } as SquashedAppUIObject);
+      const childAttributes = this.normalizeAttributes(childElement.attributes);
+      elementsArray.push({ [elementName]: childAttributes } as SquashedAppUIObject);
     });
     return elementsArray;
   }
+
+  private normalizeAttributes(attributes: XMLAttributes) {
+    const keys = Object.keys(attributes);
+    keys.forEach(key => {
+      if (typeof attributes[key] === 'string') {
+        if (key === 'bounds' && typeof attributes.bounds === 'string') {
+          const regexp = /\d+/g;
+          const boundsKeys = ['x', 'y', 'height', 'width'];
+          let match: RegExpExecArray;
+          const out = {};
+          while ((match = regexp.exec(attributes.bounds)) !== null) {
+            out[boundsKeys.shift()] = Number(match[0]);
+          }
+          attributes.bounds = out as bounds;
+        } else if (
+          key === 'children' ||
+          key === 'count' ||
+          key === 'focusItem' ||
+          key === 'index' ||
+          key === 'opacity' ||
+          key === 'loadStatus'
+        ) {
+          attributes[key] = Number(attributes[key]);
+        } else if (key === 'focusable' || key === 'focused' || key === 'visible') {
+          attributes[key] = attributes[key] === 'true';
+        }
+      }
+    });
+    return attributes;
```

Comment:
would probably make sense for some of the booleans at least to add defaults, can just do `return {
...defaultAttributes, ...attributes } `

Reply:
🤔 hmm.... not sure if I want to do that? Since this is just translating what the Roku gives us into more digestible format

Reply:
those are attributes that apply to every node, its just for some reason roku doesnt report them if they haven't changed

Reply:
focused, visibile, focusable, at least

Reply:
Is there a list of these default attributes? Wouldn't be opposed to em if we had a definite list...

Reply:
Added in 6edb702